Create vscode devcontainer to streamline dev envs (#1080)
adap/flower: doc/source/devcontainer.rst

VS Code Remote - Containers 

> ======================================

Reviewer: Remove excess `=`

> ======================================

Author: ```suggestion
===========================
```

When working on the Flower framework we want to ensure that all contributors use the same developer environment to format code or run tests. For this purpose we are using the VSCode Remote Containers extension. What is it? Read the following quote:

> `Quoting the offical VSCode documentation <https://code.visualstudio.com/docs/remote/containers>`_

Reviewer: Was this line intended to show a quote?

Author: Yes. Thats the reason why the subsequent two lines are indented. That link is only the source of the quote.

Author: Change a bit so its not confusing

>     The Visual Studio Code Remote - Containers extension lets you use a Docker container as a full-featured development environment. It allows you to open any folder inside (or mounted into) a container and take advantage of Visual Studio Code's full feature set. A devcontainer.json file in your project tells VS Code how to access (or create) a development container with a well-defined tool and runtime stack. This container can be used to run an application or to separate tools, libraries, or runtimes needed for working with a codebase.

Reviewer: ```suggestion
The Visual Studio Code Remote - Containers extension lets you use a Docker container as a fully-featured development environment. It allows you to open any folder inside (or mounted into) a container and take advantage of Visual Studio Code's full feature set. A :code:`devcontainer.json` file in your project tells VS Code how to access (or create) a development container with a well-defined tool and runtime stack. This container can be used to run an application or to separate tools, libraries, or runtimes needed for working with a codebase.
```

>     Workspace files are mounted from the local file system or copied or cloned into the container. Extensions are installed and run inside the container, where they have full access to the tools, platform, and file system. This means that you can seamlessly switch your entire development environment just by connecting to a different container.

Reviewer: ```suggestion
Workspace files are mounted from the local file system or copied or cloned into the container. Extensions are installed and run inside the container, where they have full access to the tools, platform, and file system. This means that you can seamlessly switch your entire development environment just by connecting to a different container.
```

Getting started
---------------

> Configuring and setting up the Dockerfile as well the configuration for the devcontainer can be a bit more involved. The good thing is you want have to do it. Usually it should be enough to install Docker on your system and ensure its available on your commandline. Additoinally install the `VSCode Containers Extension <vscode:extension/ms-vscode-remote.remote-containers>`_.

Reviewer: ```suggestion
Configuring and setting up the :code:`Dockerfile` as well the configuration for the devcontainer can be a bit more involved. The good thing is you want have to do it. Usually it should be enough to install Docker on your system and ensure its available on your command line. Additoinally, install the `VSCode Containers Extension <vscode:extension/ms-vscode-remote.remote-containers>`_.
```

> Now you should be good to go. When starting VSCode it will ask you to run in the container environment and if you confirm automatically build the Container use it. To manually switch to it you can click the green area in the bottom left corner of you VSCode after installing the extension and you will be able to select something along the lines of "(Re)Open Folder in Container"

Reviewer: ```suggestion
Now you should be good to go. When starting VSCode, it will ask you to run in the container environment and - if you confirm - automatically build the container and use it. To manually instruct VSCode to use the devcontainer, you can, after installing the extension, click the green area in the bottom left corner of your VSCode window and select the option *(Re)Open Folder in Container*.
```
